partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -377,11 +378,15 @@ internal class PlanTyper(
                     }
                     is Rex.Op.Var.Resolved -> root
                 }
-                val steps = item.steps
-                relOpExcludeItem(resolvedRoot, steps)
+                relOpExcludePath(resolvedRoot, path.steps)
             }
-
-            val op = relOpExclude(input, newItems)
+            val subsumedPaths = newPaths
+                .groupBy(keySelector = { it.root }, valueTransform = { it.steps }) // combine exclude paths with the same resolved root before subsumption
+                .map { (root, allSteps) ->
+                    val nonRedundant = ExcludeRepr.toExcludeRepr(allSteps.flatten()).removeRedundantSteps()
+                    relOpExcludePath(root, nonRedundant.toPlanRepr())
+                }
+            val op = relOpExclude(input, subsumedPaths)
```

Comment:
Is this in case the AST representations aren't equivalent, but perhaps they resolve to the same?

```
SELECT * EXCLUDE t.a.*, "t".a FROM { 'a': { 'x': 1 } ... } AS t
```

I _think_ the only time the root (which is a var) would differ is based on the case-sensitivity of the identifier.